contracts/Vouching.sol

```diff
@@ -1,216 +1,529 @@
-   * @param name String representing the name of the dependency.
-   * @param owner Address that owns the created dependency.
-   * @param dependencyAddress Address of the created dependency.
-   * @param initialStake uint256 representing the amount vouched in the creation of the dependency.
+   * @dev Tells the the initial minimum amount of vouched tokens a dependency can have when being created.
+   * @return A uint256 number with the minimumStake value.
    */
-  event DependencyCreated(
-    bytes32 indexed nameHash,
-    string name,
-    address indexed owner,
-    address indexed dependencyAddress,
-    uint256 initialStake
-  );
+  function minimumStake() public view returns(uint256) {
+    return minimumStake_;
+  }
 
   /**
-   * @dev Emitted when the owner of a dependency vouches for it.
-   * @param nameHash bytes32 hash of the name of the dependency.
-   * @param amount uint256 with the amount vouched on the dependency.
+   * @dev Tells the ERC20 token being used for vouching.
+   * @return The address of the ERC20 token being used for vouching.
    */
-  event Vouched(bytes32 indexed nameHash, uint256 amount);
+  function token() public view returns(ERC20) {
+    return token_;
+  }
 
   /**
-   * @dev Emitted when the owner of a dependency removes their vouch.
-   * @param nameHash bytes32 hash of the name of the dependency.
-   * @param amount uint256 with the amount of stake that has been removed from the dependency.
+   * @dev Tells the appeal payout fee.
+   * @return The appeal payout fee.
    */
-  event Unvouched(bytes32 indexed nameHash, uint256 amount);
+  function appealFee() public view returns(uint256) {
+    return appealFee_;
+  }
 
   /**
-   * @dev Emitted when a dependency has been removed from the registry.
-   * @param nameHash bytes32 nameHash hash of the name of the dependency.
+   * @dev Tells the address of the overseer.
+   * @return The address of the overseer in charge of the vouching contract.
    */
-  event DependencyRemoved(bytes32 indexed nameHash);
-
-  using SafeMath for uint256;
-  using SafeERC20 for ERC20;
-  using Address for address;
+  function overseer() public view returns(address) {
+    return overseer_;
+  }
 
   /**
-   * @dev Struct that represents a particular dependency, with its address, owner and amount of tokens vouched.
+   * @dev Tells the information associated to an entry
    */
-  struct Dependency {
-    address owner;
-    address dependencyAddress;
-    uint256 stake;
+  function getEntry(uint256 _entryID)
+    public view returns (
+      address addr,
+      address owner,
+      string metadataURI,
+      bytes32 metadataHash,
+      uint256 minimumStake,
+      uint256 totalVouched,
+      uint256 totalAvailable,
+      uint256 totalBlocked
+    )
+  {
+    if (!_existsEntry(_entryID)) return (address(0), address(0), "", bytes32(0), uint256(0), uint256(0), uint256(0), uint256(0));
+    Entry memory e = entries_[_entryID];
```

Comment:
This is because I was thinking about whether it was correct to pass a storage pointer to an internal view function, which is not supposed to change the storage status. Thus, I started copying into memory those variables before calling those functions, but, since it was definitely spending more gas than before, I finally decided to rollback. I obviously forgot to undo these changes, thanks :)

However, I would think a bit more if there is any side effect on having storage pointers mixed with view functions, just in case.